pkg/resources/table.go

```diff
@@ -101,6 +108,19 @@ var tableSchema = map[string]*schema.Schema{
 			},
 		},
 	},
+	"data_retention_days": {
+		Type:         schema.TypeInt,
+		Optional:     true,
+		Default:      1,
```

Comment:
Will this default value overwrite the inherited value from a higher level object? I believe the default behavior in snowflake is to inherit from the "higher object". For example, if I were to set `data_retention_days=20` for my schema, I'd expect that to be inherited by the table. This seems like it might break that?

Reply:
I also think since the variable is called `data_retention_time_in_days` we should also name it that here. https://docs.snowflake.com/en/sql-reference/parameters.html#data-retention-time-in-days

It does look like the `Default` value is 1 in the docs though.

Reply:
More context 
_If a retention period is specified for a database or schema, the period is inherited by default for all objects created in the database/schema_

from https://docs.snowflake.com/en/user-guide/data-time-travel.html

Reply:
@alldoami @berosen - I really wasn't sure what to use as `snowflake_schema` has `data_retention_days` and `snowflake_database` has `data_retention_time_in_days` so there's inconsistencies already.

https://github.com/chanzuckerberg/terraform-provider-snowflake/blob/main/pkg/resources/schema.go#L54
https://github.com/chanzuckerberg/terraform-provider-snowflake/blob/main/pkg/resources/database.go#L25

@berosen - yes you're right it will override the default from a parent object, I could change this so that if its nil then the setting is not applied to the table if that's more suitable. Which in the case of using this provider, it will be the schema as that has the logic I have here, e.g. it defaults to 1, rather than defaults to inherit.

@alldoami - I can change the behaviour of the `snowflake_schema` resource too, but obviously that will have some side effects for existing deployments. I have no idea why on the database the value is computed though.

Reply:
@robbruce , personally I like the idea of using `nil`. I feel the change would be more in-line with expected behaviour. I think making the change in `snowflake_schema` would be a good idea as well. However, I'll defer to whatever @alldoami feels is best.

Reply:
@berosen - I have been looking into this, and I'm not sure its possible with how the current typing system works for `schema.TypeInt` - see https://github.com/hashicorp/terraform-plugin-sdk/issues/261

Essentially , if I set Default to be `nil`, then the SDK returns this as `0` and so there doesn't appear to be a nice way to inherit.

In addition, when reading a table with `SHOW TABLES`, then even if the value was not supplied (so it inherits) then it returns the value defined in it's parent schema, so even then there is no way to detect if there is a difference.

Reply:
Ah ok I see your reasoning now with the naming @robbruce.

Reply:
@robbruce That's a good point, I think the work around for that would be complicated. Maybe it's enough to write a note in the description that the inheritance needs to be explicitly defined by passing the same value as the parent object? ie
```
snowflake_table foo {
   ...
   data_retention_days=snowflake_schema.name. data_retention_days
}
```
Sorry for the noise all!

Reply:
@berosen @alldoami - updated the description as suggested and the table example too.